packages/neo-one-client/src/user/keystore/LedgerHandler.ts

```diff
@@ -0,0 +1,148 @@
+
+  return paramsMatched.map((val) => parseInt(val, 16));
+};
+
+export class LedgerHandler {
+  public static readonly byteLimit = LedgerTransport.byteLimit;
+  public static readonly type = LedgerTransport.type;
+
+  public static readonly init = async () => {
+    const isSupported = await LedgerTransport.isSupported();
+    if (!isSupported) {
+      throw new Error(`Ledger not supported by your machine`);
+    }
+    const paths = await LedgerTransport.list();
+    if (paths.length === 0) {
+      throw new Error(`No Ledger detected on your machine`);
+    }
+    const ledger = await LedgerTransport.open(paths[0]);
+
+    return new LedgerHandler(ledger);
+  };
+
+  private readonly ledger: HWLedger;
+  private mutableQueue: LedgerQueueRequest[];
+  private mutableRunning: boolean;
+
+  public constructor(ledger: HWLedger) {
+    this.ledger = ledger;
+    this.mutableRunning = false;
+    this.mutableQueue = [];
+  }
+
+  public async getPublicKey(account = 0): Promise<string> {
+    const response = await this.send('80040000', BIP44(account), [VALID_STATUS]);
+
+    return crypto.toECPoint(response.message).toString('hex');
+  }
+
+  public async sign({ message, account }: { readonly message: string; readonly account: number }): Promise<string> {
+    const data = message + BIP44(account);
+    const chunks = data.match(/.{1,510}/g) || [];
+    const initChunks = chunks.slice(0, chunks.length - 1);
+    const finalChunk = chunks[chunks.length - 1];
+
+    await Promise.all(
+      initChunks.map(async (chunk) => {
+        await this.send(`80020000`, chunk, [VALID_STATUS]);
+      }),
+    );
+
+    const response = await this.send(`80028000`, finalChunk, [VALID_STATUS]);
+
+    return response.message.toString('hex');
+  }
+
+  public async close(): Promise<void> {
+    this.ledger.close();
+  }
+
+  private async send(params: string, msg: string, statusList: ReadonlyArray<number>): Promise<LedgerResponse> {
+    // tslint:disable-next-line:promise-must-complete
+    const promise = new Promise<LedgerResponse>((resolve, reject) => {
+      this.mutableQueue.push({
+        params,
+        msg,
+        statusList,
+        resolve,
+        reject,
+      });
+    });
+
+    this.startSendLoop();
+
+    return promise;
+  }
+
+  private async startSendLoop(): Promise<void> {
+    if (this.mutableRunning) {
+      return;
+    }
+
+    this.mutableRunning = true;
+
+    let entry = this.mutableQueue.shift();
+    // tslint:disable-next-line:no-loop-statement
+    while (entry !== undefined) {
+      try {
+        const result = await this.doSend(entry.params, entry.msg, entry.statusList);
+        entry.resolve(result);
+      } catch (error) {
+        entry.reject(error);
+      }
+      entry = this.mutableQueue.shift();
+    }
+
+    this.mutableRunning = false;
+  }
+
+  private async doSend(params: string, msg: string, statusList: ReadonlyArray<number>): Promise<LedgerResponse> {
+    const [cla, ins, p1, p2] = getParams(params);
```

Comment:
da fuq are `cla` `ins` `p1` `p2`.

Also same comment, just make these actual parameters on the `send` method.